makefile.shade

```diff
@@ -780,4 +841,138 @@ functions
 
             return reposToBuild.ToList();
         }
+
+        static IEnumerable<IGrouping<int, RepositoryInfo>> BatchReporties()
+        {
+            var repositoryLookup = new List<RepositoryInfo>();
+            foreach (var repo in GetRepositoriesToBuild())
+            {
+                var info = new RepositoryInfo { Name = repo };
+                var srcDir = Path.Combine(repo, "src");
+
+                if (Directory.Exists(srcDir))
+                {
+                    foreach (var directory in Directory.EnumerateDirectories(srcDir))
+                    {
+                        info.RepositoryNames.Add(Path.GetFileName(directory));
+                        var projectJson = Path.Combine(directory, "project.json");
+                        if (File.Exists(projectJson))
+                        {
+                            GetDependencies(projectJson, info.DependencyNames);
+                        }
+                    }
+                }
+
+                var otherDirs = new[] { "test", "samples" };
+                for (var i = 0; i < otherDirs.Length; i++)
+                {
+                    var otherDir = Path.Combine(repo, otherDirs[i]);
+                    if (Directory.Exists(otherDir))
+                    {
+                        foreach (var directory in Directory.EnumerateDirectories(otherDir))
+                        {
+                            var projectJson = Path.Combine(directory, "project.json");
+                            if (File.Exists(projectJson))
+                            {
+                                GetDependencies(projectJson, info.DependencyNames);
+                            }
+                        }
+                    }
+                }
+
+                info.DependencyNames.ExceptWith(info.RepositoryNames);
+                repositoryLookup.Add(info);
+            }
+
+            foreach (var info in repositoryLookup)
+            {
+                foreach (var item in info.DependencyNames)
+                {
+                    var dependency = repositoryLookup.Find(r => r.RepositoryNames.Contains(item));
+                    if (dependency != null)
+                    {
+                        // Testing has odd cyclic dependencies
+                        if (string.Equals(info.Name, "Testing", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
We know every repo depends on Testing and Testing depends on Common, PlatformAbstractions and Logging - how do you propose we resolve it?